stdlib/xml/dom/pulldom.pyi

```diff
@@ -1,26 +1,44 @@
-from typing import IO, Any
+from typing import IO, Any, Sequence, Tuple, Union
+from typing_extensions import Literal
+from xml.dom.minidom import Document, DOMImplementation, Element, Text
 from xml.sax.handler import ContentHandler
 from xml.sax.xmlreader import XMLReader
 
-START_ELEMENT: str
-END_ELEMENT: str
-COMMENT: str
-START_DOCUMENT: str
-END_DOCUMENT: str
-PROCESSING_INSTRUCTION: str
-IGNORABLE_WHITESPACE: str
-CHARACTERS: str
+START_ELEMENT: Literal["START_ELEMENT"]
+END_ELEMENT: Literal["END_ELEMENT"]
+COMMENT: Literal["COMMENT"]
+START_DOCUMENT: Literal["START_DOCUMENT"]
+END_DOCUMENT: Literal["END_DOCUMENT"]
+PROCESSING_INSTRUCTION: Literal["PROCESSING_INSTRUCTION"]
+IGNORABLE_WHITESPACE: Literal["IGNORABLE_WHITESPACE"]
+CHARACTERS: Literal["CHARACTERS"]
+
+DocumentFactory = Union[DOMImplementation, None]
+Node = Union[Document, Element, Text]
+
+Event = Tuple[
+    Literal[
+        Literal["START_ELEMENT"],
+        Literal["END_ELEMENT"],
+        Literal["COMMENT"],
+        Literal["START_DOCUMENT"],
+        Literal["END_DOCUMENT"],
+        Literal["PROCESSING_INSTRUCTION"],
+        Literal["IGNORABLE_WHITESPACE"],
+        Literal["CHARACTERS"],
+    ],
+    Node,
+]
 
 class PullDOM(ContentHandler):
-    document: Any | None
-    documentFactory: Any
+    document: Union[Document, None]
```

Comment:
Please use the new union syntax:

```suggestion
    document: Document | None
```

(It doesn't work yet for the type aliases above.)

Reply:
When i use the union syntax, some of the checks fail. Is that normal?